distributed/shuffle/_rechunk.py

```diff
@@ -414,19 +425,21 @@ def partial_rechunk(
             axis[index] for index, axis in zip(global_index, x.chunks)
         )
         if _slicing_is_necessary(ndslice, original_shape):
-            input_task = (slice_name,) + global_index
-            dsk[(slice_name,) + global_index] = (
+            key = (slice_name,) + ndpartial.ix + global_index
+            input_key = (slice_name,) + ndpartial.ix + global_index
+            dsk[key] = (
                 getitem,
                 (x.name,) + global_index,
                 ndslice,
             )
         else:
-            input_task = (x.name,) + global_index
+            input_key = (x.name,) + global_index
 
-        transfer_keys.append((transfer_name,) + global_index)
-        dsk[(transfer_name,) + global_index] = (
+        key = (transfer_name,) + ndpartial.ix + global_index
```

Comment:
OK, so this PR makes the partial index a "structured" part of the key, rather than an unstructured string? And this helps the scheduler figure out what depends on what?

I think this all makes sense, but perhaps a comment indicating how this key decomposition helps to describe the dependencies would be good. AFAICT we're not changing the information that is going in to the graph, but just how we structure it. Is that right?

Reply:
Yes, exactly. We move the information related to the partial out of the portion of the task key that represents the group and into the remainder of the key. I've renamed variables to make this clearer, I'm not sure how useful it is to add more documentation around key decomposition here. Though I agree that we should have a docs page somewhere that explains how groups and prefixes get inferred from various types of (composite) keys.

Reply:
I've added _minor_ commentary and will add an issue regarding some dedicated docs.